opm/simulators/wells/GroupEconomicLimitsChecker.cpp

```diff
@@ -0,0 +1,311 @@
+/*
+  Copyright 2023 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+#include <opm/simulators/wells/GroupEconomicLimitsChecker.hpp>
+#include <opm/input/eclipse/Schedule/Well/WellTestConfig.hpp>
+#include <fmt/format.h>
+
+namespace Opm {
+
+GroupEconomicLimitsChecker::
+GroupEconomicLimitsChecker(
+    const BlackoilWellModelGeneric &well_model,
+    WellTestState &well_test_state,
+    const Group &group,
+    const double simulation_time,
+    const int report_step_idx,
+    DeferredLogger &deferred_logger
+) :
+    well_model_{well_model}
+  , group_{group}
+  , simulation_time_{simulation_time}
+  , report_step_idx_{report_step_idx}
+  , deferred_logger_{deferred_logger}
+  , well_state_{well_model.wellState()}
+  , well_test_state_{well_test_state}
+  , schedule_{well_model.schedule()}
+  , gecon_props_{schedule_[report_step_idx_].gecon().get_group_prop(
+                 schedule_, well_model_.summaryState(), group_.name())}
+{
+    for (std::size_t i = 0; i < this->phase_idx_map_.size(); i++) {
+        auto phase_idx = this->phase_idx_map_[i];
+        this->phase_idx_reverse_map_[phase_idx] = static_cast<int>(i);
+        auto phase_pos = this->well_model_.phaseUsage().phase_pos[phase_idx];
+        double production_rate = WellGroupHelpers::sumWellSurfaceRates(
+            this->group_, this->schedule_, this->well_state_,
+            this->report_step_idx_, phase_pos, /*isInjector*/false);
+        this->production_rates_[i] = this->well_model_.comm().sum(production_rate);
+    }
+}
+
+/****************************************
+ * Public methods in alphabetical order
+ ****************************************/
+
+void
+GroupEconomicLimitsChecker::
+activateEndRun()
+{
+
```

Comment:
Since this functionality is missing. Please add to the list in https://github.com/OPM/opm-simulators/blob/c6e30fbf3c3a9677ec245b004ce55358b9021c40/opm/simulators/utils/PartiallySupportedFlowKeywords.cpp#L1

and a throw here. 

The same goes for workover etc. that is missing.